src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -949,6 +983,8 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			const currentTime = new Date().getTime() / 1000;
 			const maxTolerance = 2 * 60; // two minutes
 			if (expiry - currentTime < maxTolerance) {
+				account.isStale = true;
+				this._accountManagementService.accountUpdated(account);
```

Comment:
Why is this being added?

Reply:
This updates the account value in the account store because I have changed the isStale field to true

Reply:
Was this functionality that was broken before or something? Because it had the expiration check here before so that seems like the case.

If this was a separate bug you noticed while fixing the original issue then I'd suggest opening a separate PR for it so it can be reviewed by itself - doing that makes  more sure there's nothing else missed and helps both reviews be much more targeted.

Reply:
The account needs to be updated separately than the connection profile, and that was a problem for the temp tables when trying to refresh the account token